compose/config/config.py

```diff
@@ -598,7 +598,10 @@ def resolve_environment(service_dict, environment=None):
     """
     env = {}
     for env_file in service_dict.get('env_file', []):
-        env.update(env_vars_from_file(env_file))
+        try:
+            env.update(env_vars_from_file(env_file))
+        except ConfigurationError as exc:
+            log.warn('{} - ignoring'.format(exc.msg))
```

Comment:
Though unlikely, a missing file isn't the only reason a ConfigurationError could be thrown. When the path exists but is not a file (e.g. a directory), a `ConfigurationError` [will also be thrown][1]. In that case, a blocking error seems more appropriate than a passive warning.

[1]: https://github.com/akalipetis/compose/blob/651283eef96a6ab7854b3b8ee57bc25467f27f75/compose/config/environment.py#L31-L32

Instead of a `try`/`except`, perhaps we should check file existence directly.

```python
if os.path.exists(filename):
  env.update(env_vars_from_file(env_file))
else:
  log.warn('env_file {} does not exist - ignoring'.format(env_file))
```

Reply:
If @akalipetis makes this change, would this be merge-able?

Reply:
bumping this, if we make this change, would this be merge-able @nilbus?